File: workspace/judoku/src/main/java/judoku/data/Block.java
package judoku.data;

public class Block extends AbstractCellSet {
	public Block(int id) {
		super(id);
	}

	protected String getName() {
		return "block";
	}

	public String toString() {
        StringBuffer buf = new StringBuffer();
        int i=0;
    	for (Cell c: this.getCells()) {
    		buf.append(c);
    		i++;
    		if (i%3==0) buf.append("\n");
    	}

        return buf.toString();
   }
}

File: workspace/scudoku/src/main/scala/scudoku/data/Block.scala
package scudoku.data

/**
 * Java2Scala port from judoku.data.Block
 */

class Block extends CellSet("block") {
  override def toString = {
      val buf: StringBuffer = new StringBuffer();
      var i: Int = 0
      for (c <- getCells) {
        buf.append(c)
        i = i+1
        if (i%3==0) buf.append("\n")
      }

      buf.toString
  }
}